### ISSUE TITLE
Extension traits are re-exported by name

### ISSUE THREAD

**Issue by benfrankel** [bevyengine/bevy#13959]
Extension traits can be re-exported without bringing their name into scope using [underscore imports](https://doc.rust-lang.org/reference/items/use-declarations.html#underscore-imports):

```rust
pub mod prelude {
    // This brings the trait AND its name into scope. This is how bevy currently does it.
    //pub use crate::commands::BuildChildrenTransformExt;

    // This brings the trait into scope.
    pub use crate::commands::BuildChildrenTransformExt as _;
}
```

Users can still pull in the name if needed:

```rust
use bevy_transform::prelude::*;
use bevy_transform::commands::BuildChildrenTransformExt;
```

Benefits:

- Clearly mark certain re-exports as extension traits
- Reduce name pollution (not that extension traits have high-value names, but still)

**Comment by BenjaminBrienen** [comment]
> This is really interesting, I hadn't considered using `trait as _` to avoid name collisions. At a glance this also works in [preludes](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=2d58697888b02fc3ff00a4ecb0234e2f). For a future PR, could be nice to do this for extension style traits (e.g., `AppAssetExt` etc.) so the preludes aren't so heavy with names.

_Originally posted by @bushrat011899 in https://github.com/bevyengine/bevy/pull/16163#discussion_r1821705357_

**Comment by BenjaminBrienen** [comment]
I agree that `use Trait as _` should be the standard in our code. I say even for normal `use`, not just exports.

**Comment by alice-i-cecile** [comment]
See https://github.com/bevyengine/bevy/pull/17207#pullrequestreview-2535350803 for rationale / decision.
